### PR TITLE
Fix incorrect pasting images on Firefox

### LINKED ISSUE #3585: Paste from Excel in FF embeds image instead of table
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open firefox
2. Copy-paste table from this excel [simple_styles.xlsx](https://github.com/ckeditor/ckeditor4/files/3738318/simple_styles.xlsx)

### Expected result

The table is copied with appropriate styles to the editor.

### Actual result

The table is copied as an image.

## Other details

The issue is not present in 4.12.1.

* Browser: Firefox
* OS: Windows10
* CKEditor version: 4.13.0
* Installed CKEditor plugins: …

### LINKED ISSUE #3625: Content pasted from PowerPoint is converted to image
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Create some PowerPoint document.
2. Copy text from one of the slides.
3. Paste into CKEditor 4.

### Expected result

Content is pasted as a text.

### Actual result

The image of the copied content is pasted.

## Other details

The issue is probably connected with the fact that Office applications attaches screenshot of copied content to the clipboard data.

* Browser: Firefox
* CKEditor version: 4.13.0
* Installed CKEditor plugins: clipboard

### LINKED ISSUE #3634: Add CKEDITOR.plugins.clipboard.dataTransfer#getTypes method
## Type of report

Task

## Provide description of the task

There are sometimes need to check what types are actually inside pasted data. `CKEDITOR.plugins.clipboard.dataTransfer#getTypes` will do exactly that: return the array of pasted types!

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What is the proposed changelog entry for this pull request?

```
Fixed Issues:

*[#3585](https://github.com/ckeditor/ckeditor4/issues/3585): [Firefox] Fixed: Excel content is pasted as an image.
*[#3625](https://github.com/ckeditor/ckeditor4/issues/3625): [Firefox] Fixed: PowerPoint content is pasted as an image.

API Changes:

*[#3634](https://github.com/ckeditor/ckeditor4/issues/3634): Added the [`CKEDITOR.plugins.clipboard.dataTransfer#getTypes()`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_plugins_clipboard_dataTransfer.html#method-getTypes) method.
```

## What changes did you make?

I've changed the mechanism for pasting images. Now the image is pasted only when `dataTransfer` for `paste` event contains only image (so its `types` property is equal to `[ 'Files' ]`). I've also added small helper function for getting types, `CKEDITOR.plugins.clipoard.dataTransfer#getTypes()`.

Closes #3585.
Closes #3625.
Closes #3634.

Note: this PR fixes more general issue connected with pasting images alongside other content. For fix for pasting Excel/PowerPoint content when PfW is enabled see #3641.

### CODE REVIEW

**Comment by f1ames** [comment]
Rebased onto latest `master`.

File: tests/plugins/clipboard/manual/pasteimagehtml.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div id="editor">
```
Comment (f1ames) [inline tests/plugins/clipboard/manual/pasteimagehtml.html]: Shouldn't this test be ignored on all browsers but Firefox?

Comment (Comandeer) [reply tests/plugins/clipboard/manual/pasteimagehtml.html]: I didn't ignore it as we will probably use the code for Firefox also in other browsers to handle pasting images (see https://github.com/ckeditor/ckeditor4/issues/2675#issuecomment-515750101). Therefore not ignoring the test is more future-proof.

**Comment by Comandeer** [comment]
Some of tests produces `CKEDITOR.plugins.clipboard.dataTransfer` objects without the native `DataTransfer` instance. I added a check to return empty array from `CKEDITOR.plugins.clipboard.dataTransfer#getTypes()` in such case.

**Comment by f1ames** [comment]
Rebased onto latest `master`.

File: tests/plugins/clipboard/manual/pasteimagehtml.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Content is pasted as text/HTML.
+
+### Unexpected
+
+Content is pasted as image.
+
+Documents to check:
+
+* [PowerPoint one](_assets/pasteimagehtml.pptx),
+* [Excel one](_assets/pasteimagehtml.xlsx).
```
Comment (f1ames) [inline tests/plugins/clipboard/manual/pasteimagehtml.md]: This assets are missing.

**Comment by Comandeer** [comment]
Added missing assets.

**Pull request merged**